modules/basicfuncs/numeric-funcs.c

```diff
@@ -130,3 +132,156 @@ tf_num_mod(LogMessage *msg, gint argc, GString *argv[], GString *result)
 }
 
 TEMPLATE_FUNCTION_SIMPLE(tf_num_mod);
+
+
+static GString *
+_make_gstring_buffer(const LogTemplateInvokeArgs *args)
+{
+  if (args->bufs->len == 0)
+    g_ptr_array_add(args->bufs, g_string_sized_new(64));
+
+  return (GString *) g_ptr_array_index(args->bufs, 0);
+}
+
+static gboolean
+_tf_num_parse_arg_with_message(const TFSimpleFuncState *state,
+                               LogMessage *message,
+                               const LogTemplateInvokeArgs *args,
+                               gint64 *number)
+{
+  GString *formatted_template_result = _make_gstring_buffer(args);
+  gint on_error = args->opts->on_error;
+
+  log_template_format(state->argv[0], message, args->opts, args->tz,
+    args->seq_num, args->context_id, formatted_template_result);
+
+  if (!parse_number_with_suffix(formatted_template_result->str, number))
+    {
+      if (!(on_error & ON_ERROR_SILENT))
+        msg_error("Parsing failed, template function's argument is not a number",
```

Comment:
You might consider to also check ON_ERROR_DROP_MESSAGE. You could produce an empty output if an invalid entry is found. Otherwise you can ignore it, as has been done here.

See also:
https://github.com/balabit/syslog-ng/issues/868
https://github.com/balabit/syslog-ng/issues/867